src/core/api-session.utils.test.ts

```diff
@@ -0,0 +1,110 @@
+import { AppEvents, DataSourceInstanceSettings } from '@grafana/data';
+import { BackendSrv } from '@grafana/runtime';
+
+jest.mock('@grafana/runtime', () => ({
+    ...jest.requireActual('@grafana/runtime'),
+    getAppEvents: jest.fn(),
+}));
+jest.mock('./utils', () => ({
+    post: jest.fn(),
+}));
+jest.resetModules();
+
+let post, getAppEvents, ApiSessionUtils;
+let mockGetAppEvents: jest.Mock;
+let mockPost: jest.Mock;
+
+describe('ApiSessionUtils', () => {
+    let instanceSettings: DataSourceInstanceSettings;
+    let backendSrv: BackendSrv;
+    let appEvents: { publish: any; };
+    let apiSessionUtils: any;
+
+    beforeEach(async () => {
+        // Dynamically import dependencies after mocks
+        ApiSessionUtils = (await import('./api-session.utils')).ApiSessionUtils;
+        getAppEvents = (await import('@grafana/runtime')).getAppEvents;
+        post = (await import('./utils')).post;
+
+        // Reset static cache before each test
+        (ApiSessionUtils as any)._sessionCache = undefined;
+
+        mockGetAppEvents = getAppEvents as jest.Mock;
+        mockPost = post as jest.Mock;
+
+        instanceSettings = { url: 'http://api-example.com' } as DataSourceInstanceSettings;
+        backendSrv = {} as BackendSrv;
+        appEvents = { publish: jest.fn() };
+        mockGetAppEvents.mockReturnValue(appEvents);
+
+        apiSessionUtils = new ApiSessionUtils(instanceSettings, backendSrv);
+
+        mockPost.mockClear();
+        appEvents.publish.mockClear();
```

Comment:
```suggestion
        // Dynamically import dependencies after mocks
        ApiSessionUtils = (await import('./api-session.utils')).ApiSessionUtils;
        getAppEvents = (await import('@grafana/runtime')).getAppEvents;
        post = (await import('./utils')).post;

        mockGetAppEvents = getAppEvents as jest.Mock;
        appEvents = { publish: jest.fn() };
        mockGetAppEvents.mockReturnValue(appEvents);

        mockPost = post as jest.Mock;

        instanceSettings = { url: 'http://api-example.com' } as DataSourceInstanceSettings;
        backendSrv = {} as BackendSrv;
        apiSessionUtils = new ApiSessionUtils(instanceSettings, backendSrv);

        // Reset static cache before each test
        (ApiSessionUtils as any)._sessionCache = undefined;

        mockPost.mockClear();
        appEvents.publish.mockClear();
```
Can we follow this order?